### PR TITLE
Add more doc including on Tagging

### PR DESCRIPTION
**Pull request by andrerom**
cc @DominikaK 

Does:
- Add usage of tags doc page
- Adds fixes on FOS doc including mentioning that it's for internal use (for core devs)

### CODE REVIEW

**Comment by DominikaK** [comment]
@andrerom: is the intent to keep the doc both in this repo and in developer-documentation? I'm a bit concerned about keeping the two versions in sync if there are changes in the future.

**Comment by andrerom** [comment]
The intention is to follow the existing convention here _(and on some topics in the kernel)_ to have internal doc here written and maintained on changes here by someone who at the time of writing knows how it works, which you and other cable of writing end user facing doc can reuse for the higher level doc.

File: docs/fos_http_cache.md [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
-Introduced with eZ Publish 5.4, [FOSHttpCacheBundle][fos] is supported by eZ Platform, and covers the following
-features:
+_This doc explains how this bundle integrates with FOSHttpCacheBundle internally, and is aimed at core developers._
 
-* Tags purging/baning (since version 1.8)
-* Http cache purge/ban
-* User context hash
+
+eZ Platform HttpCache _(this bundle)_ uses [FOSHttpCacheBundle 1.x][fos] in the following way:
+- When `local` purge type is set, system uses a custom Symfony Proxy tag aware data store, and hence does not
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `the system uses` and `tag-aware`

File: docs/fos_http_cache.md [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+  use FOSHttpCacheBundle directly. However FOSHttpCache(Bundle) can still be used to manipulate responses, incl tagging.
+- When `http` purge type is set, system uses FOSHttpCache configured with varnish as proxy client.
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `the system`

File: docs/fos_http_cache.md [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
+
+As HttpCache in eZ Publish/Platform pre dates FOSHttpCache, we have own abstractions for:
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `predates` (one word)

File: docs/fos_http_cache.md [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
+ - purge clients: In this bundle exposed as `ezplatform.http_cache.purge_client` service
+ - App Cache: `EzSystems\PlatformHttpCacheBundle\AppCache` app cache class to extend from, this in turn extends
+   `FOS\HttpCacheBundle\SymfonyCache\EventDispatchingHttpCache` however provides own set of request aware purgers and
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `, however it provides` and `request-aware`

File: docs/fos_http_cache.md [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
+   own handling of user context hash which for BC reasons is called `X-User-Hash` in eZ.
+
+_Note: Once FOSHttpCache(Bundle) has full support for tagging we might se a major version of this bundle be refactored to more
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `Once FOSHttpCache(Bundle) has full support for tagging, a major version of this bundle might be be refactored to more`

File: docs/fos_http_cache.md [hunk 6]
```diff
@@ -1,21 +1,30 @@
...
+directly reuse FOSHttpCache(Bundle). However in the meantime our abstractions has allowed us to ship features across
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `our abstractions have`

File: docs/fos_http_cache.md [hunk 7]
```diff
@@ -1,21 +1,30 @@
...
+Symfony Cache and Varnish such as full tagging support not possible with plain FOSHttpCacheBundle._
+
+Some further differences on how HttpCache behaves on specific features in this bundle can be found below.
 
 ## Http cache clear
-Varnish proxy client from FOSHttpCache lib is now used for clearing eZ Http cache, even when using Symfony HttpCache.
+Varnish proxy client from FOSHttpCache lib is used for clearing cache when `http` purge client is configired..
```
Comment (DominikaK) [inline docs/fos_http_cache.md]: `configured`

File: docs/using_tags.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
+# Using Tags
+
+For understanding and taking the most advantage of `ezplatform-http-cache`, understanding tags is key _(pun intended)_.
```
Comment (DominikaK) [inline docs/using_tags.md]: `Understanding tags is the key _(pun intended)_ to making the most of...`

File: docs/using_tags.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+They work in a similar way as [persistence cache tags in eZ Platform v2](https://github.com/ezsystems/ezpublish-kernel/tree/7.0/doc/specifications/cache/persistence):
+- A set of secondary keys set on every cache item, on top of "primary key" which in this case is the URI
+- Like a index in a database it is typically used for anything relevant that represent the given cache item
```
Comment (DominikaK) [inline docs/using_tags.md]: `an index`

File: docs/using_tags.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+- Used for cache invalidation
+
+There is also other aspects to this bundle, like the fact by using Varnish xkey over BAN we can now do soft purge, but
```
Comment (DominikaK) [inline docs/using_tags.md]: `There are also other aspects...` and `the fact that by using`

File: docs/using_tags.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+more on that in [varnish/varnish.md](varnish/varnish.md) and the corresponding VCL file.
+
+For a bit less abstract way of saying this; It allows us to do things like tagging every article response, and when
+article content type gets an update we can tell Varnish all articles should be considered stale so they are updated in
+the background once someone requests them. Same goes for other operations by the repository.
+
+## Tags in use in this bundle
+
+- `content-<content-id>` :
+  _Used on anything that are affected on changes to content, on content itself as well as location and so on._
```
Comment (DominikaK) [inline docs/using_tags.md]: `anything that is affected by changes`

File: docs/using_tags.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+- `content-type-<content-type-id>` :
+  _For use when content type changes affecting content of it's type._
```
Comment (DominikaK) [inline docs/using_tags.md]: `its type`

File: docs/using_tags.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+- `location-<location-id>` :
+  _Used for clearing all cache relevant for a given location._
+
+- `parent-<parent-location-id>` :
+  _Useful for clearing all children of a parent, or in all siblings._
+
+- `path-<location-id>` :
+  _For operations that changes the tree itself, like move/remove/(..)._
```
Comment (DominikaK) [inline docs/using_tags.md]: `operations that change`

File: docs/using_tags.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+- `relation-<content-id>` :
+   _For use when updates affect their all reverse relations (NOTE: System does not add this tag to responses itself yet,
```
Comment (DominikaK) [inline docs/using_tags.md]: `affect all their reverse...`

File: docs/using_tags.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+
+
+### For responses with X-Location-Id
+
+For custom or eZ controllers _(like REST at the time of writing)_ still using `X-Location-Id`, a dedicated response
+listener `XLocationIdResponseSubscriber` handles translating this to tags so the cache can be properly invalidated by
+this bundle.
+
+*This is currently marked as Deprecated, and for rendering content it is thus advice to refactor to use Content View.
+For other needs there is a internal tag handler in this bundle that can be used, however be aware it will probably
```
Comment (DominikaK) [inline docs/using_tags.md]: `an internal tag handler`

File: docs/using_tags.md (hunk 8 above)
Comment (andrerom) [reply docs/using_tags.md]: skipping the joke, not everyone will get it and I don't want to waste time explaining tags is a key also inside the joke as the joke is then moot :)

File: docs/fos_http_cache.md (hunk 3 above)
Comment (bdunogier) [reply docs/fos_http_cache.md]: I think André meant "existed before". "predate" is like in "predator" is it not ? :)

File: docs/using_tags.md [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+  _Used on anything that are affected on changes to content, on content itself as well as location and so on._
+
+- `content-type-<content-type-id>` :
+  _For use when content type changes affecting content of it's type._
+
+- `location-<location-id>` :
+  _Used for clearing all cache relevant for a given location._
+
+- `parent-<parent-location-id>` :
+  _Useful for clearing all children of a parent, or in all siblings._
```
Comment (bdunogier) [inline docs/using_tags.md]: "clearing cache of all the children of a location"

File: docs/fos_http_cache.md (hunk 3 above)
Comment (andrerom) [reply docs/fos_http_cache.md]: I did, but is it predator? :) :P

**Comment by andrerom** [comment]
@DominikaK @bdunogier updated

**Comment by DominikaK** [comment]
"predate" actually does mean "to exist earlier", but if the first association is with predator, then maybe it's not the most fortunate word here ;)
I'd vote for HTTP cache, in all caps, because it is an abbreviation after all, and most places including W3 spell it like this.

**Comment by vidarl** [comment]
LGTM

But PR description says "Updates VCL to reflect what is in ezplatform" .... It does not seem like the PR touched any .vcl files

**Comment by bdunogier** [comment]
> But PR description says "Updates VCL to reflect what is in ezplatform" .... It does not seem like the PR touched any .vcl files

Good point from @vidarl.

> "predate" actually does mean "to exist earlier", but if the first association is with predator, then maybe it's not the most fortunate word here ;)

Well, it is indeed what first came to my mind.

Predator

**Comment by andrerom** [comment]
> But PR description says "Updates VCL to reflect what is in ezplatform" .... It does not seem like the PR touched any .vcl files

right, those syncs went in in 2ffb3f08927ac3f6d3585cb4fa38ed3e269fbf61 and 436fdc6037841f73beccce83bbb3ee467bf837c0

**Pull request merged**